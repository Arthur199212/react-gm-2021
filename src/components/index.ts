export * from './App'
export * from './Button'
export * from './Card'
export * from './CardMedia'
export * from './CloseIcon'
export * from './DeleteMovieForm'
export * from './Dropdown'
export * from './DropdownItem'
export * from './DualRingSpinner'
export * from './ErrorBoundary'
export * from './Filters'
export * from './Footer'
export * from './FormField'
export * from './GenreFormField'
export * from './InputBox'
export * from './Modal'
export * from './MovieCard'
export * from './MovieForm'
export * from './NoSearchResults'
export * from './SearchResults'
export * from './ShowMoreButton'
export * from './SmallModal'
export * from './SuccessIcon'
export * from './Tabs'
export * from './TabItem'
export * from './TextareaBox'
export * from './TextareaFormField'
