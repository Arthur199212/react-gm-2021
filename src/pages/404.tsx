export { NotFoundPage as default } from '@app/features'
