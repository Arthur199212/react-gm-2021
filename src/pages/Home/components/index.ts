export * from './Filters'
export * from './NoSearchResults'
export * from './SearchResults'
