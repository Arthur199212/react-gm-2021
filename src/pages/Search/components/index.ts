export * from './Header'
export * from './SearchBox'
