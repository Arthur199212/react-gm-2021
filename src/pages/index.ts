export * from './Movie'
export * from './NotFound'
export * from './Search'
