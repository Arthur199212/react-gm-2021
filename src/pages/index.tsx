import { NextPageContext } from 'next'
import { useRouter } from 'next/router'

const IndexPage = () => {
  const router = useRouter()

  // Make sure we are in the browser
  if (typeof window !== 'undefined') {
    router.push('/search')
  }

  return <></>
}

export const getStaticProps = (ctx: NextPageContext) => {
  // We check for ctx.res to make sure we are on the server
  if (ctx.res) {
    ctx.res.writeHead(302, { Location: '/search' })
    ctx.res.end()
  }
  return { props: {} }
}

export default IndexPage
