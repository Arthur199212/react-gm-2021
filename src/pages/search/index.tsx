import { NextPageContext } from 'next'
import { fetchMoviesDefaultThunk } from '@app/features/SearchPage/store'
import { createStore } from '@app/store'

export { SearchPage as default } from '@app/features'

export const getServerSideProps = async (props: NextPageContext) => {
  const store = createStore()
  await store.dispatch(fetchMoviesDefaultThunk())

  return { props: { preloadedState: store.getState() } }
}
