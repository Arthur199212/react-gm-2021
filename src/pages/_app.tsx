import type { AppProps } from 'next/app'
import Head from 'next/head'
import { Provider } from 'react-redux'
import { ErrorBoundary } from '@app/components'
import { createStore } from '@app/store'
import 'normalize.css'
import '@app/styles/index.scss'

const App = ({ Component, pageProps }: AppProps) => {
  const store = createStore(pageProps.preloadedState)

  return (
    <>
      <Head>
        <title>Movie Search App</title>
      </Head>
      <Provider store={store}>
        <ErrorBoundary>
          <Component {...pageProps} />
        </ErrorBoundary>
      </Provider>
    </>
  )
}

export default App
