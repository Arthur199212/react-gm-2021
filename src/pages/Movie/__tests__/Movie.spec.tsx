import React from 'react'
import { BrowserRouter as Router } from 'react-router-dom'
import { render, screen } from '@testing-library/react'
import { FiltersTestIds, SearchResultsTestIds } from '@app/components'
import { Movie } from '@app/pages'

describe('Movie Component', () => {
  it('should render properly', () => {
    render(
      <Router>
        <Movie />
      </Router>
    )

    expect(screen.getByTestId(FiltersTestIds.CONTAINER)).toBeInTheDocument()
    expect(screen.getByTestId(SearchResultsTestIds.CONTAINER)).toBeInTheDocument()
  })
})
