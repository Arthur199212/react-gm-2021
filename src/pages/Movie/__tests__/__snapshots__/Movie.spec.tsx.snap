// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Movie Page should render properly 1`] = `[Function]`;
